### ISSUE TITLE
View specific code could be moved to views

### ISSUE THREAD

**Issue by Jbarget** [dwyl/tudo#160]
at the moment we've got [`name_from_auth`](https://github.com/dwyl/tudo/blob/authentication/web/models/user_from_auth.ex#L24-L36) in the `UserFromAuth` model, I remember when going through the phoenix book that usually the logic for rendering user information into a view is held in the view layer: `views/page_views.ex`

https://github.com/dwyl/learn-phoenix-framework/blob/master/rumbrella/apps/rumbl/web/views/user_view.ex

@des-des what you thinking?

**Comment by des-des** [comment]
Hmm ill have a think about this tomorrow, when ill be looking to the auth branch again, but it sounds reasonable

**Comment by des-des** [comment]
@Jbarget yeah I see what you are saying. I am not sure it is as clear cut as that.

We are taking that data that auberauth / gh gives our application, and extracting the information we are interested in, to put inside our session object https://github.com/dwyl/tudo/pull/163/files#diff-ddd7f4f3d995e8dba8f020ca22593eeeR21.
Ie we want a session to hold a username.

I think the idea here is to build out a very basic model (that does not really do anything) that could in the future could be extended to manage users in a db if we decided to persist any user data.

In the context of our application, the only place we are using the username is in a view, so we could move that logic into the view.

What do we do if we want to put the oauth_token into the session, that logic would not want to live inside the view, but would happily fit alongside our current user from auth stuff.

In short, helpers that are only concerned with transforming data for a template can live inside a view, but logic that transforms some data to create a session should not live in a view.

I our case, I think we could move that functionality into the controller, or attach the auth object we have in our callback function as a session, then move the trasformation code into a helper in the view.

I do not think the last option there is good

**Comment by Jbarget** [comment]
I get what you're saying and agree 👍
